crates/tests/signer-tests/src/lib.rs

```diff
@@ -17,29 +22,39 @@ fn ensure_env_variables() -> Result<()> {
                 "Environment variables not set and failed to load signer.env: {}",
                 e
             )
-        })
+        })?;
+
+    dotenvy::from_filename("node.env").map(|_| ()).map_err(|e| {
+        anyhow!(
+            "Environment variables not set and failed to load node.env: {}",
+            e
+        )
+    })
 }
 
-async fn get_signer_channel() -> Result<Channel> {
+async fn connect_with_timeout(port_env_var: &str, label: &str) -> Result<Channel> {
     ensure_env_variables()?;
-    let signer_port = std::env::var("GRPC_PORT")?;
-
-    let address = format!("https://localhost:{}", signer_port);
-
+    let port = env::var(port_env_var)?;
+    let address = format!("https://localhost:{}", port);
     let timeout = Instant::now() + Duration::from_secs(3);
-    let channel = loop {
-        if let Ok(c) = tonic::transport::Channel::builder(address.parse()?)
-            .connect()
-            .await
-        {
-            break c;
-        }
-        if Instant::now() > timeout {
-            return Err(anyhow!("timeout waiting for signer"));
+
+    loop {
+        match Channel::builder(address.parse()?).connect().await {
+            Ok(channel) => return Ok(channel),
+            Err(_) if Instant::now() > timeout => {
+                return Err(anyhow!("timeout waiting for {}", label));
+            }
+            Err(_) => tokio::time::sleep(Duration::from_millis(100)).await,
         }
-    };
+    }
```

Comment:
YOu chanched the implementation, but it adds nothing. Except the sleep, but you can just add it a the bottom of the loop without changing the rest of the logic